### DISCUSSION TITLE
Use Dependency Graph to inform `Last Modified` Dates

### DISCUSSION THREAD

**Discussion by jalberto** [11ty/eleventy#4168] [Enhancement Queue]
**Describe the bug**
when using `date: Last Modified` in a `liquid` file that `includes` other files, the date doesn't change when only an include is being modified

**To Reproduce**
- create a file with 3 includes
- modify any of the includes but not the parent file
- you get the old date

**Expected behavior**
`Last-Modified` should take into consideration compiled result for date

**Environment:**

- Linux
- Eleventy Version 0.10.0

**Comment by uncenter** [comment]
This would require Eleventy to analyze what templates are included/rendered (dependencies) in order to use those date(s) as the last modified date. Related to https://github.com/harttle/liquidjs/issues/233, though that solution (rendering the template with a fake filesystem to capture read calls) does not seem viable.

**Comment by zachleat** [comment]
Blocked by #4167 (for Liquid specifically)
